### ISSUE TITLE
Transition.EndNavigation not always triggered 

### ISSUE THREAD

**Issue by oomek** [mickelson/attract#586]
Let assume we want to remap the navigation buttons for a certain layout without remapping default controls in settings. When we intercept all the signals `up`  `down`  `left`  `right`  and call `next_game` and `prev_game` we have the following output in the log:

```
Command intercepted by script handler: up
Command intercepted by script handler: prev_game
[Transition] type=EndNavigation, var=0
Command intercepted by script handler: down
Command intercepted by script handler: next_game
[Transition] type=EndNavigation, var=0
Command intercepted by script handler: left
Command intercepted by script handler: prev_game
Command intercepted by script handler: right
Command intercepted by script handler: next_game
```

As you can see the `EndNavigation` transition is only triggered for buttons that are mapped in Controls as `next_game` and `prev_game`

Here is a minimalistic code that is showing the problem

```
function on_signal( sig )
{
	switch ( sig )
	{
		case "left":
			fe.signal( "prev_game" )
			return true

		case "right":
			fe.signal( "next_game" )
			return true

		case "up":
			fe.signal( "prev_game" )
			return true

		case "down":
			fe.signal( "next_game" )
			return true

		case "prev_game":
		case "next_game":
			return true

		default:
			return false
	}
}
fe.add_signal_handler( "on_signal" )
```

**Comment by oomek** [comment]
The same goes for changing index:

```
[Transition] type=ToNewSelection, var=-1
[Transition] type=FromOldSelection, var=1
[Transition] type=EndNavigation, var=0
Command intercepted by script handler: up
[Transition] type=EndNavigation, var=0

[Transition] type=ToNewSelection, var=1
[Transition] type=FromOldSelection, var=-1
[Transition] type=EndNavigation, var=0
Command intercepted by script handler: down
[Transition] type=EndNavigation, var=0

[Transition] type=ToNewSelection, var=-1
[Transition] type=FromOldSelection, var=1
[Transition] type=EndNavigation, var=0
Command intercepted by script handler: left

[Transition] type=ToNewSelection, var=1
[Transition] type=FromOldSelection, var=-1
[Transition] type=EndNavigation, var=0
Command intercepted by script handler: right
```

For `up` and `down` the EndNavigation is called after each `fe.list.index` change and then when the button is released

For left and right it's called only after `fe.list.index` change

```
function on_signal( sig )
{
	switch ( sig )
	{
		case "left":
			fe.list.index--
			return true

		case "right":
			fe.list.index++
			return true

		case "up":
			fe.list.index--
			return true

		case "down":
			fe.list.index++
			return true

		case "prev_game":
		case "next_game":
			return true

		default:
			return false
	}
}
fe.add_signal_handler( "on_signal" )

I don't know yet what would be the best way around it. Maybe a new Squirrel function that would allow to remap the default signals for the direction controls ( those in brackets )?

**Comment by oomek** [comment]
Conclusions:
Current implementation only allows vertical layouts benefit from `EndNavigation` transition. If you have mixed layouts, some vertical and some horizontal, remapping signals in controls is not an option. Additionally some end-users are unaware of that neccessity.

Shouldn't the `EndNavigation` transition be called only after releasing of the key after changing the value of `fe.list.index` ?

**Comment by oomek** [comment]
```
fe.remap_control( "left", "prev_game" )
fe.remap_control( "right", "next_game" )
fe.remap_control( "up", "prev_filter" )
fe.remap_control( "down", "next_filter" )
```

Just a rough idea. It would be temporary if the layout sets it.

**Comment by oomek** [comment]
I prepared a branch with a new function I've added `fe.set_action()`
https://github.com/oomek/attract/commit/70e9d0f927d9c3910aba1ca2dabbe92710b88dd8

I'm not opening a PR yet as we are still testing it under all scenarios.

**Comment by mickelson** [comment]
I think I've got a fix for this

**Comment by oomek** [comment]
Not neccessarily, if you just suppress the signal without calling `next_game` the `EndNavigation transition now fires up every time.
Besides this does not help for horizontal layouts. Have you checked my branch yet?

**Comment by oomek** [comment]
With My commit we can take advantage of the navigation acceleration without remapping actions every time you change the layout.

**Comment by keilmillerjr** [comment]
🤦‍♂️

**Comment by oomek** [comment]
Hats off to your commit https://github.com/mickelson/attract/commit/a45bff88e413e81ee8c38ff3e48fd3c7b0c74aae

It actually fixed the underlying problem. I have to admit that my commit was a workaround as I've decided to not to mess with the main loop too much.

By adding just:
```
case "left":
    fe.signal( "prev_game" )
    return true

case "right":
    fe.signal( "next_game" )
    return true
```
we have now proper transitions calling and acceleration.
Closing the issue. Thanks Andrew.
